cucumber-spring/src/main/java/io/cucumber/spring/TestContextAdaptor.java

```diff
@@ -53,6 +53,7 @@ private void notifyTestContextManagerAboutBeforeTestMethod() {
         try {
             Class<?> delegateTestClass = delegate.getTestContext().getTestClass();
             delegateTestInstance = applicationContext.getBean(delegateTestClass);
+            delegate.prepareTestInstance(delegateTestInstance);
```

Comment:
Note to self: `delegateTestInstance` is used as an implicit condition to determine if `beforeTestMethod` has fired. Consider separating creation+preparation from assignemtn.